pkg/cluster/backup_manager.go

```diff
@@ -91,6 +92,11 @@ func (bm *backupManager) setupStorage() (s backupstorage.Storage, err error) {
 			return nil, errNoS3ConfigForBackup
 		}
 		s, err = backupstorage.NewS3Storage(c.S3Context, c.KubeCli, cl.Name, cl.Namespace, *b)
+	case spec.BackupStorageTypeABS:
```

Comment:
@hongchaodeng in addition, I seem to have lost clarity on the path of storage backend configuration through the code on its way to actually creating backups.  

For instance, what exactly is being set up in this stanza and how does it compare/tie-in with the sidecar deployment setup further below?  I found myself thinking perhaps we only need the sidecar deployment setup (`k8sutil.AttachABSToPodSpec(..)`) but alas the etcd cluster resource can't be created if I remove the logic here...

Reply:
The logical storage components looks like this:

- backup manager: This sits on operator. It abstracts backup sidecar related logic (k8s)  for creating deployments, and backup storage specific logic (S3/ABS) for cloning data for restore and so on.
- backup sidecar: This is the backup sidecar that only takes care of saving/serving snapshots.

So basically the storage logic has been split up into two places. We can't abstract away all logic into backup sidecar.